cylc/flow/scripts/scan.py

```diff
@@ -278,6 +282,23 @@ def state_totals_key():
     )
 
 
+def format_and_write(write, formatter, *f_args, **f_kwargs):
+    """Wrapper for the various stdout formatters.
+
+    Format and write out scanned info, or warn of corrupted contact files
+    (which have correct name and path, but are missing required content).
+    """
+    try:
+        write(formatter(*f_args, **f_kwargs))
+    except KeyError:
```

Comment:
This is catching a bit more than just missing contact file fields, I think the formatters only use host & port from the contact file, might worth shifting the try/except down a level or possibly even `.get(..., '?')`.

Reply:
I was trying to avoid doing this in each individual formatter.

Do you mean the `write` function could potentially raise `KeyError`?  Or, I suppose the "rich" formatter could potentially raise `KeyError` for non contact-file reasons (missing data fields from the server).

Reply:
> for non contact-file reasons (missing data fields from the server).

Yeah, that sort of thing, in which case the real error would get swallowed and masked by a contact file error.